main/tasks/create_jobs_task.c

```diff
@@ -35,30 +35,28 @@ void create_jobs_task(void *pvParameters)
             (GLOBAL_STATE->ASIC_functions.set_version_mask)(GLOBAL_STATE->version_mask);
             GLOBAL_STATE->new_stratum_version_rolling_msg = false;
         }
-        ESP_LOGI(TAG, "New Work Dequeued %s", mining_notification->job_id);
-
-        // Process this job immediately
-        process_mining_job(GLOBAL_STATE, mining_notification);
 
-        // Now wait for more work or process additional jobs if needed
-        uint32_t iteration_count = 0;
+        ESP_LOGI(TAG, "New Work Dequeued %s", mining_notification->job_id);
         while (GLOBAL_STATE->stratum_queue.count < 1 && GLOBAL_STATE->abandon_work == 0)
         {
-            // Check if we need to generate more work based on the current job
             if (should_generate_more_work(GLOBAL_STATE))
             {
-                generate_additional_work(GLOBAL_STATE, mining_notification);
+                generate_work(GLOBAL_STATE, mining_notification, extranonce_2);
```

Comment:
@skot https://github.com/skot/ESP-Miner/pull/478/commits/26d7f1913fc1c7cbbaf5c5f43205e5b7da6e440f

Do we need for extanoce_2 to get reset per mining.notify, or can we restart the calculation for each notify? e.g.

The current implementation basically works like this.
> 0 -> 1 -> 2 -> Notify -> 0 -> 3 -> 4 -> Notify -> 0 -> 5 -> 6 etc 

The new implementation can either work
> 0 -> 1 -> 2 -> Notify -> 3 -> 4 -> 5 -> Notify -> 6 -> 7 -> 8 etc 
or
> > 0 -> 1 -> 2 -> Notify -> 0 -> 1 -> 2 -> Notify -> 0 -> 1 -> 2 etc

Since extanoce_1 is provided with a different value I personally prefer 
> > 0 -> 1 -> 2 -> Notify -> 0 -> 1 -> 2 -> Notify -> 0 -> 1 -> 2 etc

Reply:
AFAIK it doesn't make any difference, other than what your OCD tells you is prettiest.

Reply:
Rolled back to 
> 0 -> 1 -> 2 -> Notify -> 0 -> 1 -> 2 -> Notify -> 0 -> 1 -> 2 etc
Thanks.